lib/katello/tasks/upgrade_check.rake

```diff
@@ -0,0 +1,36 @@
+DESIRED_VERSION = "2.4.2"
+SYSTEM = "Katello"
+CP_LISTEN_ACTION = "Actions::Candlepin::ListenOnCandlepinEvents"
+
+def current_version
+  Katello::VERSION
+end
+
+namespace :katello do
+  task :upgrade_check => ['environment'] do
+    desc "Task that can be run before upgrading to Katello 3.0 to check if it's upgrade ready"
+    success = "PASS"
+    fail = "FAIL"
+
+    puts "This script makes no modifications and can be re-run multiple times for the most up to date results."
+    puts "Checking #{SYSTEM} upgradeability...\n\n"
+
+    # check for any running tasks
+    task_count = ::ForemanTasks::Task.active.where("label != '#{CP_LISTEN_ACTION}'").count
+    task_status = task_count > 1 ? fail : success
+    puts "Checking for running tasks..."
+    puts "[#{task_status}] - There are #{task_count} active tasks.\n\n"
+
+    # check the current version
+    version_status = current_version == DESIRED_VERSION ? success : fail
+    puts "Checking the current version..."
+    puts "[#{version_status}] - Current version of #{SYSTEM} is #{current_version} " \
+      "and needs to be #{DESIRED_VERSION}\n\n"
```

Comment:
Nah, I spent a while thinking about this and tried a few different things but could never come up with a way to solve it how they requested in the BZ.  You can leave it, I suppose.  It lets the user know when we expected them to run the script I guess.